lib/net/hostlist.js

```diff
@@ -1072,10 +1078,10 @@ class HostList {
       return addrs;
     }
 
-    const {port, key} = target;
+    const {port} = target;
```

Comment:
@pinheadmz do we need to use `target` here? Whether `target` is a node address or dns seed domain, I think we want to set the resultant `NetAddress`'s port to `this.network.defaultPort` and its key to `null`.

Reply:
Good point, I posted a commit (we can squash) that simplifies the process.

It's weird because we use the seeds list for both DNS seeds and actual peers. The latter require a custom port and a fallback port, but DNS seeds only ever use one port (53) and the results they return will always have no port (A/AAAA records) so we must add one.

Bcoin still runs this way: assigning the default network port to the DNS seed, and then the resolved records all inherit that port -- I'm not sure why that is still in place when we can simplify it like in https://github.com/handshake-org/hsd/pull/364/commits/296e89751fd6420bfffbca0abf8b9172f1b9aca7.

Then of course adding Brontide complicates it more which is why we ended up with the bug. If we ever set up something like BOLT-10 for Brontide peers, we may need to adjust this whole thing again, maybe even set up DNS seeds on a non-standard port.